include/onnxruntime/core/framework/op_kernel_context.h

```diff
@@ -214,21 +217,20 @@ class OpKernelContext {
 #endif
 
   // Creates the OrtValue* based on the shape, if it does not exist
-  OrtValue* OutputMLValue(int index, const TensorShape& shape);
+  virtual OrtValue* OutputMLValue(int index, const TensorShape& shape);
+
+  virtual OrtValue* GetOrCreateOutputMLValue(int index);
 
  private:
   ORT_DISALLOW_COPY_AND_ASSIGNMENT(OpKernelContext);
-
-  OrtValue* GetOrCreateOutputMLValue(int index);
-
   int GetInputArgIndex(int index) const;
   int GetImplicitInputArgIndex(int index) const;
   int GetOutputArgIndex(int index) const;
 
-  IExecutionFrame* const execution_frame_;
-  const OpKernel* const kernel_;
-  concurrency::ThreadPool* const threadpool_;
-  const logging::Logger* const logger_;
+  IExecutionFrame* const execution_frame_{nullptr};
```

Comment:
Just a note, you can just use {} to initialize to nullptr (less typing and less to read)

https://en.cppreference.com/w/cpp/language/zero_initialization